client/src/main/java/io/pravega/client/state/SynchronizerConfig.java

```diff
@@ -41,8 +51,63 @@
      */
     int readBufferSize;
     
-    public static class SynchronizerConfigBuilder {
+    public static class SynchronizerConfigBuilder implements ObjectBuilder<SynchronizerConfig> {
         private EventWriterConfig eventWriterConfig = EventWriterConfig.builder().retryAttempts(Integer.MAX_VALUE).enableConnectionPooling(true).build();
         private int readBufferSize = 256 * 1024;
     }
+
+    static class SynchronizerConfigSerializer
+            extends VersionedSerializer.WithBuilder<SynchronizerConfig, SynchronizerConfigBuilder> {
+        @Override
+        protected SynchronizerConfigBuilder newBuilder() {
+            return builder();
+        }
+
+        @Override
+        protected byte getWriteVersion() {
+            return 0;
+        }
+
+        @Override
+        protected void declareVersions() {
+            version(0).revision(0, this::write00, this::read00);
+        }
+
+        private void read00(RevisionDataInput revisionDataInput, SynchronizerConfigBuilder builder) throws IOException {
+            builder.readBufferSize(revisionDataInput.readInt());
+            builder.eventWriterConfig.fromBytes(ByteBuffer.wrap(revisionDataInput.readArray()));
```

Comment:
This line here is the problem. `fromBytes` is a static method that returns a new event writer config. Here it should call the static method and then set the resulting EventWriterConfig on the builder. Right now it is just throwing it away.